tests/common/test_logger.py

```diff
@@ -161,34 +174,336 @@ def test_periodic_forwards_message_and_args(self, mock_info, mock_error, mock_wa
+        mock.assert_called_with(logger.LogLevel.WARNING, "--unit-test-WARNING--")
+        mock.reset_mock()
+
+        appender.write(logger.LogLevel.ERROR, "--unit-test-ERROR--")
+        mock.assert_called_with(logger.LogLevel.ERROR, "--unit-test-ERROR--")
+        mock.reset_mock()
 
+        appender.write(logger.LogLevel.INFO, "--unit-test-INFO--")
+        mock.assert_not_called()
+        mock.reset_mock()
+
+        for i in range(5):
+            appender.write(logger.LogLevel.ERROR, "--unit-test-ERROR--")
+            appender.write(logger.LogLevel.INFO, "--unit-test-INFO--")
+
+        self.assertEqual(5, mock.call_count)  # Only ERROR should be called.
+
+    @patch("azurelinuxagent.common.event.send_logs_to_telemetry", return_value=True)
     @patch('azurelinuxagent.common.event.EventLogger.save_event')
-    def test_telemetry_logger1(self, mock_save):
+    def test_telemetry_logger_sending_correct_fields(self, mock_save, patch_conf_get_logs_to_telemetry):
         appender = logger.TelemetryAppender(logger.LogLevel.WARNING, add_log_event)
-        appender.write(logger.LogLevel.WARNING, "--unit-test--")
+        appender.write(logger.LogLevel.WARNING, 'Cgroup controller "memory" is not mounted. '
+                                                'Failed to create a cgroup for extension '
+                                                'Microsoft.OSTCExtensions.DummyExtension-1.2.3.4')
 
         self.assertEqual(1, mock_save.call_count)
         telemetry_json = json.loads(mock_save.call_args[0][0])
 
-        self.assertEqual('FFF0196F-EE4C-4EAF-9AA5-776F622DEB4F', telemetry_json['providerId'])
-        self.assertEqual(7, telemetry_json['eventId'])
+        self.assertEqual(TELEMETRY_LOG_PROVIDER_ID, telemetry_json['providerId'])
+        self.assertEqual(TELEMETRY_LOG_EVENT_ID, telemetry_json['eventId'])
 
         self.assertEqual(12, len(telemetry_json['parameters']))
         for x in telemetry_json['parameters']:
             if x['name'] == 'EventName':
                 self.assertEqual(x['value'], 'Log')
-
             elif x['name'] == 'CapabilityUsed':
                 self.assertEqual(x['value'], 'WARNING')
-
             elif x['name'] == 'Context1':
-                self.assertEqual(x['value'], '--unit-test--')
-
+                self.assertEqual(x['value'], 'Cgroup controller "memory" is not mounted. '
+                                             'Failed to create a cgroup for extension '
+                                             'Microsoft.OSTCExtensions.DummyExtension-1.2.3.4')
             elif x['name'] == 'Context2':
                 self.assertEqual(x['value'], '')
-
             elif x['name'] == 'Context3':
                 self.assertEqual(x['value'], '')
+
+    @patch('azurelinuxagent.common.event.EventLogger.save_event')
+    def test_telemetry_logger_not_on_by_default(self, mock_save):
+        appender = logger.TelemetryAppender(logger.LogLevel.WARNING, add_log_event)
```

Comment:
this test should be done for all the appenders, not only the telemetry appender

Reply:
This test was validating specifically if Telemetry Appender is sending values or not. 

If I understand your concern, do you mean to test - just doing a log.warn should not generate any kind of log calls?

Reply:
Added a new test - `test_no_appenders_added` which validates no logs are written in any appender if no appender is added.